app/controllers/api/v1/coffeeshops_controller.rb

```diff
@@ -1,9 +1,26 @@
 class Api::V1::CoffeeshopsController < ApplicationController
+    protect_from_forgery unless: -> { request.format.json? }
+    
     def index
         render json: Coffeeshop.all
     end
 
     def show
         render json: Coffeeshop.find(params[:id])
     end
-end
\ No newline at end of file
+
+    def create
+        coffeeshop = Coffeeshop.new(coffeeshop_params)
+
+        if coffeeshop.save
+            render json: coffeeshop
+        else
+
+            render json: { error: coffeeshop.errors.full_messages }, status: :unprocessable_entity
+        end
+    end
+
+    def coffeeshop_params 
+        params.require(:coffeeshop).permit(:name, :address, :city, :state, :zip, :description)
+    end
```

Comment:
Great catch! There was a transition when we had a discussion about formatting etiquette for the group, and we agreed to set tab size 2. This file was unaffected since it was made before the convo.